app/controllers/api/timeslots_controller.rb

```diff
@@ -15,9 +15,11 @@ def create
   end
 
   def update
-    @timeslot = Timeslot.find_by(id: safe_params[:id])
-    @timeslot.student_id = current_user.id
-    if @timeslot.save
+    timeslot = Timeslot.find_by(id: safe_params[:id])
+    timeslot.student_id = current_user.id
```

Comment:
i would include control flow here to relay a message or redirect in case some how the current user is also the tutor.